### DISCUSSION TITLE
Unable to Access Original Markdown After Transformation in Markmap v17

### DISCUSSION THREAD

**Discussion by eldertek** [markmap/markmap#247] [Q&A]
Hello I am working on a project using Markmap to transform and visualize Markdown content. In Markmap v14, the transformation output provided a detailed structure that included information about the headings, their levels, and the lines they corresponded to, making it easier to trace back to the original Markdown content.

For example, in v14, the transformation output looked like this:
```{
    "t": "heading",
    "d": 1,
    "p": {
        "lines": [
            0,
            1
        ]
    },
    "v": "Title 1",
    "c": [
        {
            "t": "heading",
            "d": 2,
            "p": {
                "lines": [
                    1,
                    2
                ]
            },
            "v": "Subtitle 1",
            "c": [
                {
                    "t": "heading",
                    "d": 3,
                    "p": {
                        "lines": [
                            4,
                            5
                        ]
                    },
                    "v": "Sub-subtitle 1.1"
                }
            ]
        },
        {
            "t": "heading",
            "d": 2,
            "p": {
                "lines": [
                    7,
                    8
                ]
            },
            "v": "Subtitle 2"
        },
        // more items...
    ]
}
```
However, in Markmap v17, the transformation output format has changed to:

```{
    "content": "Title 1",
    "children": [
        {
            "content": "Subtitle 1",
            "children": [
                {
                    "content": "Sub-subtitle 1.1",
                    "children": [],
                    "payload": {
                        "lines": "4,5"
                    }
                }
            ],
            "payload": {
                "lines": "1,2"
            }
        },
        {
            "content": "Subtitle 2",
            "children": [],
            "payload": {
                "lines": "7,8"
            }
        },
        // more items...
    ],
    "payload": {
        "lines": "0,1"
    }
}
```
While this format is more concise, I am unable to easily access and display the original Markdown content in my application.

I would like to find a way to access the original Markdown content after transformation in Markmap v17. Specifically, I need to display both the transformed content and the original Markdown in my application.

**Comment by eldertek** [comment]
Especially for equation, the content element don't contain the markdown but html...

**Comment by gera2ld** [comment]
How about storing the original Markdown somewhere and finding the corresponding content by `payload.lines`?

BTW, markmap is not lossless, so you cannot convert the generated data back to the original Markdown.

**Reply by eldertek** [reply]
Worked ! I don't want markmap to be not lossless, but maybe like it was before, storing both orinal and converted ?
It's creepy to search lines etc.
